src/code/z_actor.c

```diff
@@ -221,180 +221,192 @@ void Actor_ProjectPos(PlayState* play, Vec3f* src, Vec3f* xyzDest, f32* cappedIn
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_MISC
+    { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },     // ACTORCAT_BOSS
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_DOOR
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_CHEST
 };
 
+NaviColor sNaviColorUnused = { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } };
+
 // unused
 Gfx D_80115FF0[] = {
     gsSPEndDisplayList(),
 };
 
-void func_8002BE64(TargetContext* targetCtx, s32 index, f32 arg2, f32 arg3, f32 arg4) {
-    targetCtx->arr_50[index].pos.x = arg2;
-    targetCtx->arr_50[index].pos.y = arg3;
-    targetCtx->arr_50[index].pos.z = arg4;
-    targetCtx->arr_50[index].unk_0C = targetCtx->unk_44;
+void Target_SetLockOnPos(TargetContext* targetCtx, s32 index, f32 x, f32 y, f32 z) {
+    targetCtx->lockOnReticles[index].pos.x = x;
+    targetCtx->lockOnReticles[index].pos.y = y;
+    targetCtx->lockOnReticles[index].pos.z = z;
+    targetCtx->lockOnReticles[index].radius = targetCtx->lockOnRadius;
 }
 
-void func_8002BE98(TargetContext* targetCtx, s32 actorCategory, PlayState* play) {
-    TargetContextEntry* entry;
-    NaviColor* naviColor;
+void Target_InitLockOn(TargetContext* targetCtx, s32 actorCategory, PlayState* play) {
+    LockOnReticle* reticle;
+    NaviColor* naviColorEntry;
     s32 i;
 
-    Math_Vec3f_Copy(&targetCtx->targetCenterPos, &play->view.eye);
-    targetCtx->unk_44 = 500.0f;
-    targetCtx->unk_48 = 0x100;
-
-    naviColor = &sNaviColorList[actorCategory];
-
-    entry = &targetCtx->arr_50[0];
-    for (i = 0; i < ARRAY_COUNT(targetCtx->arr_50); i++) {
-        func_8002BE64(targetCtx, i, 0.0f, 0.0f, 0.0f);
-        entry->color.r = naviColor->inner.r;
-        entry->color.g = naviColor->inner.g;
-        entry->color.b = naviColor->inner.b;
-        entry++;
-    }
-}
-
-void Actor_SetNaviToActor(TargetContext* targetCtx, Actor* actor, s32 actorCategory, PlayState* play) {
-    NaviColor* naviColor = &sNaviColorList[actorCategory];
-    targetCtx->naviRefPos.x = actor->focus.pos.x;
-    targetCtx->naviRefPos.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
-    targetCtx->naviRefPos.z = actor->focus.pos.z;
-    targetCtx->naviInner.r = naviColor->inner.r;
-    targetCtx->naviInner.g = naviColor->inner.g;
-    targetCtx->naviInner.b = naviColor->inner.b;
-    targetCtx->naviInner.a = naviColor->inner.a;
-    targetCtx->naviOuter.r = naviColor->outer.r;
-    targetCtx->naviOuter.g = naviColor->outer.g;
-    targetCtx->naviOuter.b = naviColor->outer.b;
-    targetCtx->naviOuter.a = naviColor->outer.a;
-}
-
-void func_8002C0C0(TargetContext* targetCtx, Actor* actor, PlayState* play) {
-    targetCtx->arrowPointedActor = NULL;
-    targetCtx->targetedActor = NULL;
-    targetCtx->unk_40 = 0.0f;
-    targetCtx->unk_8C = NULL;
+    Math_Vec3f_Copy(&targetCtx->lockOnPos, &play->view.eye);
+    targetCtx->lockOnRadius = 500.0f;
+    targetCtx->lockOnAlpha = 0x100;
```

Comment:
We do colors and alpha in dec instead of hex usually
```suggestion
    targetCtx->lockOnAlpha = 256;
```

Reply:
256 is not in the 0-255 range though 🤔

Reply:
This one is a bit weird. The actual alpha value used is a local variable in `Target_Draw`, which defaults to 255. 
The value in the actor context variable only gets used as its fading out (when `lockOnActor` is null). The temp variable stores the actor context variable - 120, and decrements by that amount every frame until it fades out (so 256 is never used, it will be 136 the first frame its used).

But, this variable having a non zero value *also* controls `Target_Draw` from running all of its logic in the first place.

Strictly speaking the value only influences the alpha of the reticle when its fading out, except for the case when its 0 and it stops drawing all together (used to turn off the reticle when link is tracking the boomerang, for example). 

Not sure if its worth indicating this nuance somehow in its name.

Reply:
my only idea is `fadeAlphaControl`
where "fade" lets you know its the fade out alpha, and "control" lets you know that it can control the reticle drawing in the first place. either way some comments are needed